### ISSUE TITLE
z88dk-lib: add target's third party libraries to include and link search paths

### ISSUE THREAD

**Issue by aralbrec** [z88dk/z88dk#390]
At the moment, third party libraries' location are not added explicitly to the include and link search paths during compiles.

This means a library named "foo" would have its header included like this in source code `#include <lib/zx/foo.h>`.  Besides being wordy, the path includes the target's name.  For libraries intended to be installable for multiple targets, this means source code would have to be edited before programs using them could be compiled.

On the compile line, the path has to be explicitly added to find the third party as in `-llib/zx/foo`.  Again the target name is there.

To remedy this, these search paths could be added to the target's cfg file.  Then includes would look like `#include <foo.h>` and compile lines would include `-lfoo`.  We can arrange to have the third party search paths added after the system search paths so that name collisions resolve in favour of the system.

**Comment by feilipu** [comment]
Yes. Noticing this more and more as a limitation, as I'm building for `rc2014`, `cpm`, `hbios`, `yaz180`, `scz180`, and need to have a long `#if __TARGET #include <arch/target.h>` set of includes at the start of the source, and then manage different compile lines.
